### PR TITLE
CARF-491: Added conversion from frontend address details to backend api model

### PR DESCRIPTION
**Pull request by maykenworthy**
- Added backend `AddressDetails` model
- Added conversion from `AddressDetailsFrontend` to `AddressDetails`
  - Includes handling for moving townOrCity up to highest available address line
- Added relevant tests

### CODE REVIEW

File: app/uk/gov/hmrc/carfregistration/models/requests/AddressDetails.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.carfregistration.models.requests
+
+import play.api.libs.json.{Json, OFormat}
+
+import scala.collection.immutable.Seq
+
+case class AddressDetails(
```
Comment (simondrugan16) [inline app/uk/gov/hmrc/carfregistration/models/requests/AddressDetails.scala]: Please rename to AddressDetailsApi for easy differentiation from AddressDetailsFrontend

File: test/models/requests/AddressDetailsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.requests
+
+import base.SpecBase
+import uk.gov.hmrc.carfregistration.models.requests.{AddressDetails, AddressDetailsFrontend}
+
+class AddressDetailsSpec extends SpecBase {
```
Comment (simondrugan16) [inline test/models/requests/AddressDetailsSpec.scala]: pls add one extra test for address line 3 -> address line 2 + townOrCity -> address line 3 if address line 2 is empty

File: test/models/requests/AddressDetailsApiSpec.scala [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          postalCode = None,
+          countryCode = "GB"
+        )
+
+        val result = AddressDetailsApi.apply(testAddressDetailsFrontend)
+
+        result mustEqual expectedResult
+      }
+
+      "when addressLine2 is empty and addressLine3 is filled, addressLine2 is populated by addressLine3 and addressLine3 is populated by townOrCity" in {}
```
Comment (simondrugan16) [inline test/models/requests/AddressDetailsApiSpec.scala]: {} at end of test name, probs not running properly? love the naming though :D

**Review by simondrugan16 — APPROVED** [review]
